### ISSUE TITLE
Universal-Node: Features missing (get_mediainfo...)

### ISSUE THREAD

**Issue by Markus-** [hklages/node-red-contrib-sonos-plus#43]
Hi,

I want to be able to create a Subflow where I put in the playername and do the "calculation" inside the subflow. By doing so I can reduce the complexity a lot.

I think there are missing some features in the Universal node.
I check the state and do different stuff if the state is playing / paused / others. Do you plan to extend the Universal node or do I miss some features of any other node which could help me to implement the logic independent of the control node?

Regards
Markus

My Flow looks like this:
```
[
    {
        "id": "cf17fc4b.2395f",
        "type": "change",
        "z": "f5499d80.2f698",
        "name": "Get STATE",
        "rules": [
            {
                "t": "set",
                "p": "payload",
                "pt": "msg",
                "to": "get_state",
                "tot": "str"
            }
        ],
        "action": "",
        "property": "",
        "from": "",
        "to": "",
        "reg": false,
        "x": 370,
        "y": 900,
        "wires": [
            [
                "8a31dd7d.6e43f"
            ]
        ]
    },
    {
        "id": "8a31dd7d.6e43f",
        "type": "sonos-get-status",
        "z": "f5499d80.2f698",
        "confignode": "a8e8ea1b.3d38d8",
        "name": "",
        "x": 510,
        "y": 900,
        "wires": [
            [
                "bbf615a5.d52218"
            ]
        ]
    },
    {
        "id": "aa85f5df.9d66b8",
        "type": "inject",
        "z": "f5499d80.2f698",
        "name": "",
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "x": 220,
        "y": 960,
        "wires": [
            [
                "cf17fc4b.2395f"
            ]
        ]
    },
    {
        "id": "bbf615a5.d52218",
        "type": "switch",
        "z": "f5499d80.2f698",
        "name": "Playing / Paused / Other",
        "property": "payload",
        "propertyType": "msg",
        "rules": [
            {
                "t": "eq",
                "v": "playing",
                "vt": "str"
            },
            {
                "t": "eq",
                "v": "paused",
                "vt": "str"
            },
            {
                "t": "else"
            }
        ],
        "checkall": "true",
        "repair": false,
        "outputs": 3,
        "x": 710,
        "y": 900,
        "wires": [
            [
                "bb8d777b.96df38"
            ],
            [
                "fe26418e.54b86"
            ],
            [
                "c5ad50ca.62436"
            ]
        ]
    },
    {
        "id": "bb8d777b.96df38",
        "type": "change",
        "z": "f5499d80.2f698",
        "name": "Pause",
        "rules": [
            {
                "t": "set",
                "p": "payload",
                "pt": "msg",
                "to": "pause",
                "tot": "str"
            }
        ],
        "action": "",
        "property": "",
        "from": "",
        "to": "",
        "reg": false,
        "x": 1070,
        "y": 880,
        "wires": [
            [
                "43f99d64.09ecf4"
            ]
        ]
    },
    {
        "id": "fe26418e.54b86",
        "type": "change",
        "z": "f5499d80.2f698",
        "name": "Play@15",
        "rules": [
            {
                "t": "set",
                "p": "payload",
                "pt": "msg",
                "to": "play",
                "tot": "str"
            },
            {
                "t": "set",
                "p": "volume",
                "pt": "msg",
                "to": "15",
                "tot": "num"
            }
        ],
        "action": "",
        "property": "",
        "from": "",
        "to": "",
        "reg": false,
        "x": 1080,
        "y": 920,
        "wires": [
            [
                "43f99d64.09ecf4"
            ]
        ]
    },
    {
        "id": "c5ad50ca.62436",
        "type": "change",
        "z": "f5499d80.2f698",
        "name": "GET MEDIAINFO",
        "rules": [
            {
                "t": "set",
                "p": "payload",
                "pt": "msg",
                "to": "get_mediainfo",
                "tot": "str"
            }
        ],
        "action": "",
        "property": "",
        "from": "",
        "to": "",
        "reg": false,
        "x": 730,
        "y": 960,
        "wires": [
            [
                "e049a7b6.5539e8"
            ]
        ]
    },
    {
        "id": "43f99d64.09ecf4",
        "type": "sonos-control-player",
        "z": "f5499d80.2f698",
        "confignode": "a8e8ea1b.3d38d8",
        "name": "SONOS-FI",
        "x": 1350,
        "y": 860,
        "wires": [
            [
                "95c51677.3d2b98"
            ]
        ]
    },
    {
        "id": "df9999f5.1b5ef8",
        "type": "switch",
        "z": "f5499d80.2f698",
        "name": "Queue Activated?",
        "property": "queueActivated",
        "propertyType": "msg",
        "rules": [
            {
                "t": "true"
            },
            {
                "t": "false"
            }
        ],
        "checkall": "true",
        "repair": false,
        "outputs": 2,
        "x": 910,
        "y": 1000,
        "wires": [
            [
                "fe26418e.54b86"
            ],
            [
                "ca7d257d.0cd968"
            ]
        ]
    },
    {
        "id": "e049a7b6.5539e8",
        "type": "sonos-get-status",
        "z": "f5499d80.2f698",
        "confignode": "a8e8ea1b.3d38d8",
        "name": "",
        "x": 730,
        "y": 1000,
        "wires": [
            [
                "df9999f5.1b5ef8"
            ]
        ]
    },
    {
        "id": "95c51677.3d2b98",
        "type": "debug",
        "z": "f5499d80.2f698",
        "name": "",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "true",
        "targetType": "full",
        "x": 1550,
        "y": 860,
        "wires": []
    },
    {
        "id": "ca7d257d.0cd968",
        "type": "change",
        "z": "f5499d80.2f698",
        "name": "Activate Queue@15",
        "rules": [
            {
                "t": "set",
                "p": "payload",
                "pt": "msg",
                "to": "activate_queue",
                "tot": "str"
            },
            {
                "t": "set",
                "p": "volume",
                "pt": "msg",
                "to": "15",
                "tot": "num"
            }
        ],
        "action": "",
        "property": "",
        "from": "",
        "to": "",
        "reg": false,
        "x": 1110,
        "y": 1000,
        "wires": [
            [
                "82824506.6fcd48"
            ]
        ]
    },
    {
        "id": "82824506.6fcd48",
        "type": "sonos-manage-queue",
        "z": "f5499d80.2f698",
        "confignode": "a8e8ea1b.3d38d8",
        "name": "Mg. Queue SONOS-FI",
        "x": 1380,
        "y": 920,
        "wires": [
            [
                "95c51677.3d2b98"
            ]
        ]
    },
    {
        "id": "a8e8ea1b.3d38d8",
        "type": "sonos-config",
        "z": "",
        "name": "SONOS-FI",
        "serialnum": "",
        "ipaddress": "10.10.0.18"
    }
]
```

**Comment by hklages** [comment]
What about a simple: Universal node toggle.playbackstate to replace the flow? 
Latest on Thursday I will publish 3.1 with lots of more commands for the Universal mode. A replacement for get_songmedia (now get.trackplus) is included.

**Comment by Markus-** [comment]
This does not work in case the queue is not active.
I have a Playlist which is played from the last paused position. If the playlist "is played" to the end the playback stopped and the queue has to be activated manually.

Or does the toggle.playbackstate reactivate the queue?

**Comment by hklages** [comment]
You can put the queue in "endless playing" mode. Then you don't have to reactivate. 

I just published 3.1. There is now also get.state (returning playback state and queue state). You can also use that.

**Comment by hklages** [comment]
... timed out. Create a new issue if you need any additional support
